presto-main/src/main/java/com/facebook/presto/operator/aggregation/state/MultiKeyValuePairStateSerializer.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.operator.aggregation.state;
+
+import com.facebook.presto.operator.aggregation.MultiKeyValuePairs;
+import com.facebook.presto.spi.block.Block;
+import com.facebook.presto.spi.block.BlockBuilder;
+import com.facebook.presto.spi.function.AccumulatorStateSerializer;
+import com.facebook.presto.spi.type.Type;
+import com.facebook.presto.type.MapType;
+
+public class MultiKeyValuePairStateSerializer
+        implements AccumulatorStateSerializer<MultiKeyValuePairsState>
+{
+    private final MapType mapType;
+
+    public MultiKeyValuePairStateSerializer(Type keyType, Type valueType)
+    {
+        this.mapType = new MapType(keyType, valueType);
+    }
+
+    @Override
+    public Type getSerializedType()
+    {
+        return mapType;
+    }
+
+    @Override
+    public void serialize(MultiKeyValuePairsState state, BlockBuilder out)
+    {
+        if (state.get() == null) {
+            out.appendNull();
+        }
+        else {
+            //  FIXME: This assumes MapType allows duplicate keys when writing object out
```

Comment:
maybe there should be a `MultimapType` with this contract written in java doc? It could be blank class inheriting from `MapType` if `MapType` currently handles correctly duplicated keys? 

Or maybe serialize it as list of entries?